tests/lms/models/group_info_test.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+from sqlalchemy.exc import IntegrityError
+
+from lms.models import ApplicationInstance, GroupInfo
+
+
+class TestGroupInfo:
+    def test_persist_and_retrieve(self, application_instance, db_session, group_info):
+        db_session.add(group_info)
+
+        retrieved_group_info = db_session.query(GroupInfo).one()
+
+        assert retrieved_group_info.id
+        assert (
+            retrieved_group_info.authority_provided_id == "test_authority_provided_id"
+        )
+        assert retrieved_group_info.consumer_key == "test_consumer_key"
+
+    def test_application_instance_relation(
+        self, application_instance, db_session, group_info
+    ):
+        db_session.add(group_info)
+
+        retrieved_group_info = db_session.query(GroupInfo).one()
+
+        assert retrieved_group_info.application_instance == application_instance
+
+    def test_authority_provided_id_cant_be_None(
+        self, application_instance, db_session, group_info
+    ):
+        group_info.authority_provided_id = None
+        db_session.add(group_info)
+
+        with pytest.raises(
+            IntegrityError, match='"authority_provided_id" violates not-null constrain'
+        ):
+            db_session.flush()
+
+    def test_application_instance_cant_be_None(
+        self, application_instance, db_session, group_info
+    ):
+        group_info.application_instance = None
+        db_session.add(group_info)
+
+        with pytest.raises(
+            IntegrityError, match='"consumer_key" violates not-null constrain'
+        ):
+            db_session.flush()
+
+    def test___str__(self, db_session, group_info):
+        db_session.add(group_info)
+        db_session.flush()
+        assert (
+            str(group_info)
+            == "<GroupInfo authority_provided_id:'test_authority_provided_id' consumer_key:'test_consumer_key'>"
+        )
+
+    def test___repr__(self, db_session, group_info):
+        db_session.add(group_info)
+        db_session.flush()
+        assert (
+            repr(group_info)
+            == "<lms.models.GroupInfo authority_provided_id:'test_authority_provided_id' consumer_key:'test_consumer_key'>"
```

Comment:
Should we be asserting so much about the `__str__` and `__repr__` methods? Seems like it might be better to assert they have some things in them we care about without being quite so prescriptive. Could be a nice use of the string matcher.

Is this to hit coverage?

Reply:
It's just so that `__str__()` and `__repr__()` are in fact tested. It think it probably tests exactly the right amount to be honest: we do actually care that it prints the correct `authority_provided_id` ('test_authority_provided_id') for example, not just that it prints any string